proto/message_contents/proof.proto

```diff
@@ -0,0 +1,32 @@
+// Signature is a generic structure for public key signatures.
+syntax = "proto3";
+
+package xmtp.message_contents;
+
+import "message_contents/signature.proto";
+
+option go_package = "github.com/xmtp/proto/v3/go/message_contents";
+option java_package = "org.xmtp.proto.message.contents";
+
+// Association Proof proves that identity key material is associated with
+// a specific individually.
+message AssociationProof {
+    // Legacy: "XMTP : Create Identity ..." Signature Payload
+    message LegacyCreateIdentityProof {
```

Comment:
Is there an advantage to storing the association "validatorType" as proto message rather than an enum?

Seems like every message would have text, signature fields

Reply:
Either way would work! I guess if we add all the fields below (SignatureFormat, SignatureGeneration, SignatureValidation), things can get clunky if we try to keep them between Proof types. Keeping things separate gives us the most clarity but also makes the code slightly clunkier.